src/bionics.cpp

```diff
@@ -970,22 +971,42 @@ bool player::install_bionics( const itype &type, int skill_level )
     }
 
     const int pk = get_painkiller();
+    const int overall_pk_dur = ( get_effect_dur(effect_pkill1) + get_effect_dur(effect_pkill2) +
+                                 get_effect_dur(effect_pkill3) + get_effect_dur(effect_pkill_l) ) / MINUTES( 1 );
     int pain_cap = 100;
     if( has_trait( trait_id( "PAINRESIST_TROGLO" ) ) ) {
         pain_cap = pain_cap / 2;
     } else if( has_trait( trait_id( "PAINRESIST" ) ) ) {
         pain_cap = pain_cap / 1.5;
     }
 
+    int fa_level = get_skill_level( skill_firstaid );
+
+    if( has_trait( trait_PROF_MED ) ) {
+        fa_level =+5;
+    }
+
     if( !has_trait( trait_id( "NOPAIN" ) ) && !has_trait( trait_id( "CENOBITE" ) ) ) {
         if( pk == 0 ) {
             popup( _( "You need to take painkillers to make installing bionics tolerable." ) );
             return false;
         } else if( pk < pain_cap / 2 ) {
-            popup( _( "You need to be a lot more numb to tolerate installing bionics.  Note that painkillers you've already taken might not be fully working yet." ) );
+            if( fa_level < 2 ) {
+                popup( _( "You need to be a lot more numb to tolerate installing bionics.  Note that painkillers you've already taken might not be fully working yet." ) );
```

Comment:
Suggest changing the final clause to, "note that painkillers youve already taken could take up to an hour to achieve full effect."